launch_testing/README.md

```diff
@@ -159,6 +159,19 @@ launch_test examples/args.test.py dut_arg:=value
 
 See the [launch_testing example with arguments](examples/args.test.py) for further reference.
 
+## ROS_DOMAIN_ID Isolation
+
+If the ROS_DOMAIN_ID environment variable isn't set, `launch_test` will automatically coordinate with other `launch_test` processes running on the same host
+to use a unique ROS_DOMAIN_ID for the launched processes.
+This allows multiple instances to run in parallel (the default with `colcon test`).
+Note that `launch_test` cannot coordinate unique domains across multiple hosts.  
+If the ROS_DOMAIN_ID environment variable is already set, `launch_test` respects the environment variable and won't attempt to select a different ID.
+In this case it's the responsibility of the user to design tests that can be safely run in parallel, or not use parallel test workers.
```

Comment:
@tfoote What if launch_testing detected that other launch_testing processes were running and printed a warning if ROS_DOMAIN_ID ls set?

I don't know of a platform independent way to do that that doesn't require the installation of additional python packages

Reply:
> What if launch_testing detected that other launch_testing processes were running and printed a warning if ROS_DOMAIN_ID ls set?

That would not be a reliable approach since a process might be about to be started but not running yet and that case wouldn't be captured.

Reply:
@dirk-thomas It's a warning, though - it would probably work well enough to tell the user that there's a problem with what they're trying to do.

Reply:
@dirk-thomas Furthermore, if you implemented this as a file lock (linux), or a named mutex (windows) where launch_test locks holds a resource for the duration, you don't have a start-up race.

Reply:
Should the current patch be merged as is or do you want to keep iterating on it? (Note: distro freeze for the Dashing release is tomorrow.)

Reply:
@dirk-thomas We should merge this - I don't have a good platform independent way to implement the above right now